lib/msf/ui/console/command_dispatcher/core.rb

```diff
@@ -2288,7 +2286,11 @@ def option_values_dispatch(o, str, words)
   # Provide valid payload options for the current exploit
   #
   def option_values_payloads
-    return @cache_payloads if @cache_payloads
+    if @cache_payloads && active_module == @previous_module
+      return @cache_payloads
+    else
```

Comment:
Since a true condition returns from the method early, there's no need to have an `else` here. The next line can go with the rest of the method body.

Reply:
Oops, good catch. We'll still need to set `@previous_module`.